### PR TITLE
Repo subdir

### PR DESCRIPTION
**Pull request by nizomovs**
Addresses #124:
- adds `subdir` property for `repo` instances

### CODE REVIEW

File: simexpal/schemes/experiments.json [hunk 1]
```diff
@@ -171,6 +171,7 @@
 								"extensions": {"$ref": "#/definitions/str_list"},
 								"extra_args": {"$ref": "#/definitions/str_list"},
 								"set": {"$ref": "#/definitions/str_or_str_list"},
+								"subdir": {"type": "string"},
```
Comment (c-bebop) [inline simexpal/schemes/experiments.json:174]: Please document that this `subdir` is now available for in our [instances readthedocs documentation](https://simexpal.readthedocs.io/en/latest/instances.html).

File: simexpal/base.py [hunk 2]
```diff
@@ -750,13 +750,17 @@ def repo_name(self):
...
+	@property
+	def instance_dir(self):
+		return os.path.join(self._cfg.instance_dir(), self._inst_yml.get('subdir', ''))
+
 	@property
 	def git_subdir(self):
 		return self._inst_yml.get('git_subdir', None)
 
 	def check_available(self):
 		for file in self.filenames:
```
Comment (c-bebop) [inline simexpal/base.py:762]: Here you should be able to do something like:
```python
files_available = [os.path.isfile(os.path.join(self.instance_dir, file)) for file in self.filenames]
return all(files_available)
```

**Review by c-bebop — CHANGES_REQUESTED** [review]

**Review by c-bebop — DISMISSED** [review]

File: docs/instances.rst [hunk 3]
```diff
@@ -40,17 +40,32 @@ Local Instances
 To add local instances to the ``instances`` key, we add a list of dictionaries with two keys
```
Comment (bernlu) [inline docs/instances.rst:40]: two keys and an optional third one

**Review by bernlu — CHANGES_REQUESTED** [review]
do we have tests that check the changes in base.py? I am not an expert on that part of the project and cannot review those changes for correctness.

File: docs/instances.rst (hunk 3 above)
Comment (nizomovs) [reply docs/instances.rst:40]: I've fixed the docs entry.

There's also a new test to check the functionality of the `instdir` key and the instance `subdir` keys. The instance files are just empty files and the test checks if all instances are available.

**Review by bernlu — APPROVED** [review]

**Review by c-bebop — APPROVED** [review]

**Pull request merged**